lib/grammers-client/src/types/photo_sizes.rs

```diff
@@ -287,10 +390,19 @@ pub struct PathSize {
 pub trait VecExt {
     /// Helper method to get the largest photo thumb
     fn largest(&self) -> Option<&PhotoSize>;
+
+    /// Helper method to get the specific type of photo thumb
+    ///
+    /// <https://core.telegram.org/api/files#image-thumbnail-types>
+    fn get_type(&self, photo_type: &str) -> Option<&PhotoSize>;
```

Comment:
I'm not entirely sold on this. How often is this needed, and why can't the user `iter().find()`?

IMO `largest()` is justified because that's the most common scenario.

This API isn't so nice, with a string that must be a single character as input, and documented outside. Feels like it's exposing even more of the raw API, a direction I'd prefer to avoid.

Reply:
It's ok, I understand your style. I'll revert this.

Reply:
By the way, what about change `photo_type` from `String` to a enum, so that user can use that enum easily without going to outside document?

Reply:
Then we need to worry about Telegram potentially removing or adding more stuff.